### ISSUE TITLE
LZ184 don't learn ladders

### ISSUE THREAD

**Issue by sysnotdown** [gcp/leela-zero#1976]
I play about 10 games with LZ184 on QQ game.  my account is lowest level, so system give me lowest level opponents.  I think the win rate should 100% because the opponents are very weak.

But tonight one man catched the leela's weakness, ladders, leela just get in the trap without doubt. winrate drops from 96% to 1% and failed. 

only about 10 games, the weakness have been catched by a lowest level man, is that right?

![default](https://user-images.githubusercontent.com/11597569/47856684-c038c380-de22-11e8-98ec-a29b89934869.PNG)

command arguments:

-g --timemanage fast --noponder -p 1000 -w D:\leela-zero-0.15-cpuonly-win64\networks\2da87ea8da0f54e87b70159e6bb82811b61d1c31091b6e019fbe62aeaa803b9c

**Comment by roy7** [comment]
Likely it just needed more playouts to properly read the ladder out? You might want to try -p 1600 since that is what selfplay is based on currently.

**Comment by alreadydone** [comment]
nit: selfplay is with -v 1600. 
@sysnotdown Do you really get 1000 playouts with a 40b net on CPU? (Within whatever the time limit was; did you change time_settings in leelaz?) With two threads (the default) I get less than 400 playouts with leelaz 0.16 release on i7-8700. I think you should at least use more threads (as many as the number of cores). 15b nets are maybe friendlier to CPUs.

**Comment by sysnotdown** [comment]
> 
> 
> nit: selfplay is with -v 1600 (weaker than -p 1600).
> @sysnotdown Do you really get 1000 playouts with a 40b net on CPU? (Within whatever the time limit was; did you change time_settings in leelaz?) With two threads (the default) I get less than 400 playouts with leelaz 0.16 release on i7-8700. I think you should at least use more threads (as many as the number of cores). 15b nets are maybe friendlier to CPUs; you may want to refer to #1914.

the time is limited to 3 seconds, because the player is very weak. maybe it limits playouts?

**Comment by alreadydone** [comment]
Yeah, I got 400 playouts **in one minute** on CPU, so I imagine you got something on the order of 20 playouts in 3 seconds, and it's not surprising that LZ misses ladders under such low playouts.

**Comment by sysnotdown** [comment]
> 
> 
> Yeah, I got 400 playouts **in one minute** on CPU, so I imagine you got something on the order of 20 playouts in 3 seconds, and it's not surprising that LZ misses ladders under such low playouts. Try add --benchmark to your commands to see how long it takes to finish and how many playouts per second (n/s) you get.

32 n/s

**Comment by alreadydone** [comment]
So you get about 100 playouts in 3 seconds, which is a lot better than the computer on which I tested, but still too low for reading ladders. You may try to add -t 8 to use more threads and/or use a 15x192 net on zero.sjeng.org. Don't you have a GPU?

**Comment by sysnotdown** [comment]
> So you get about 100 playouts in 3 seconds, which is a lot better than the computer on which I tested, but still too low for reading ladders. Don't you have a dedicated GPU? On CPU you may try to add -t 8 to use more threads and/or use a 192x15 net on zero.sjeng.org. You may also want to join one of the QQ groups here: https://www.goandai.com/leela-zero/#1529460399562-8425459e-e663

no GPU, only 48 cpus. I want to know how many playouts can read ladders? thanks!

**Comment by alreadydone** [comment]
@sysnotdown I never tested and I think it depends on the situation.
